src/xunit.performance.api/Profilers/Etw/AddressSpace.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Xunit.Performance.Api.Profilers.Etw
+{
+    /// <summary>
+    /// Represents an address space as given by the Microsoft.Diagnostics.Tracing.Parsers
+    /// </summary>
+    internal sealed class AddressSpace : ISpan<ulong>
+    {
+        /// <summary>
+        /// Initializes a new instance of the AddressSpace class.
+        /// </summary>
+        /// <param name="address">Interval's start address.</param>
+        /// <param name="size">Interval's size.</param>
+        public AddressSpace(ulong address, int size)
+        {
+            Start = address;
+            Size = size;
+            End = Start + (uint)Size;
+        }
+
+        /// <summary>
+        /// The size of the address space.
+        /// </summary>
+        public int Size { get; }
```

Comment:
`Size` will always be >= 0, so does `uint` or `ulong` make sense here?

Reply:
I started with the same interface provided by the `TraceEvent` library. I can change it now.

Reply:
I started with the same interface provided by the `TraceEvent` library. I can change it now.

---
In reply to: [145194543](https://github.com/Microsoft/xunit-performance/pull/242#discussion_r145194543) [](ancestors = 145194543)